UAF-50 UAF-542 Add Fund Group, Sub-fund and Object Sub-type to Org Review Document

leahasullivan opened the pull request:
Created several files that override the delivered foundation code that include the implementation of adding these three new fields to the Org Review Document.  There are new properties that need to be added to the institutional-config.properties file that will allow the new files to be picked up by the code.  There are Liquibase updates that were added to update the database.

leahasullivan commented:
Conflict Resolution changes: (Most were from ticket UAF-43 which was the objectSubType Routing ticket)

spring-coa.xml: added the bean needed for the AccountingOrganizationHierarchyReviewRoleTypeServiceImpl
KfsKimAttrubtes.java: needed to add in the variables, getters/setters for fundGroupCode as objectSubType were already in ua-development from UAF-43
db.changelog-master.xml: added in the needed include for the liquibase for UAF-50.
spring-sys.xml: no additions, the code was just ordered differently
KfsKimAttributes.xml: needed to add the fundGroupCode information to the existing code from UAF-43
spring-purap.xml: no additions, the code was just ordered differently
institutional-config.properties:  code was ordered differently, plus added in the new class paths for this ticket.
FinancialSystemDataDictionaryBaseTypes.xml: needed to add the additions for fundGroupCode and the routing for OrganizationHierarchy. Plus updated the routing attribute value for objectSubTypeCode as it was named differently in UAF-43 than for this modification.

Additional changes needed because of the name of the bean was different after resolving merge conflicts:
OrgReviewRole.xml: updated the parent bean value for OrgReviewRole-financialObjectSubTypeCode-parentBean
PurchasingAccountsPayableDataDictionaryBaseTypes.xml: updated the routing attribute for the objectSubTypeCode.
EffortCertificationDocument.xml: updated the routing attribute for the objectSubTypeCode.

Retested the modification after changes and everything has passed.

Perrinek commented on kfs-ec/src/main/resources/edu/arizona/kfs/module/ec/document/datadictionary/EffortCertificationDocument.xml:
```diff
@@ -0,0 +1,40 @@
+  			<value>overrideCode</value>
+  			<value>account.subFundGroup.fundGroupCode</value>
+  			<value>account.subFundGroupCode</value>
+  			<value>objectCode.financialObjectSubTypeCode</value>
```
Looking at the EffortCertificationDetail.java class, I would think this should be:
<value>financialObject.financialObjectSubTypeCode</value>

Perrinek commented on kfs-core/src/main/java/edu/arizona/kfs/coa/document/OrgReviewRoleMaintainableImpl.java:
```diff
@@ -0,0 +1,45 @@
+import org.kuali.rice.kns.web.ui.Field;
+
+public class OrgReviewRoleMaintainableImpl extends org.kuali.kfs.coa.document.OrgReviewRoleMaintainableImpl {
+
```
This code was not included in the original modification, but I think it was missed:
 protected void prepareFieldsCommon(Field field, boolean shouldReviewTypesFieldBeReadOnly, boolean hasAccountingOrganizationHierarchy){
...
if(OrgReviewRole.REVIEW_ROLES_INDICATOR_FIELD_NAME.equals(field.getPropertyName())) {
            field.setReadOnly(true);
        } else if( !hasAccountingOrganizationHierarchy
                && (OrgReviewRole.FROM_AMOUNT_FIELD_NAME.equals(field.getPropertyName()) ||
                        OrgReviewRole.TO_AMOUNT_FIELD_NAME.equals(field.getPropertyName()) ||
Add new fields here ---->
                        OrgReviewRole.OVERRIDE_CODE_FIELD_NAME.equals(field.getPropertyName()))) {
            field.setReadOnly(true);
        }
This is so that only documents that have the AccountingOrganizationHierarchy routing node can use the new fields. Please let me know if you want to discuss.

ghost commented:
I have completed my initial code review.

leahasullivan commented:
Changed the document value to financialObject.financialObjectSubTypeCode in the EffortCertificationDocument.xml and added the three new fields to the prepareFieldsCommon method in the OrgReviewRoleMaintainableImpl.java.

ghost commented:
Looks good!
